src/ray/core_worker/core_worker.cc

```diff
@@ -607,6 +615,9 @@ Status CoreWorker::Delete(const std::vector<ObjectID> &object_ids, bool local_on
   absl::flat_hash_set<ObjectID> memory_object_ids;
   GroupObjectIdsByStoreProvider(object_ids, &plasma_object_ids, &memory_object_ids);
 
+  // TODO(edoakes): what are the desired semantics for deleting from a non-owner?
+  // Should we just delete locally or ping the owner and delete globally?
```

Comment:
I wonder if we'd really need the Delete/free interface with ref counting. Manual deletion doesn't interact well with the ref counting and in theory, you could just use `del x_id` in python to decrement the ref count and get the same effect.

Reply:
Agreed, I think that would make sense. Given that we only expose this in the internal API I think we should consider deprecating it.